### PR TITLE
[e2e] Refactor InterpreterOperation retain test with resourceinterpreter

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

As talk in https://github.com/karmada-io/karmada/pull/2839#discussion_r1029150199.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/resourceinterpreter_test.go [hunk 1]
```diff
@@ -81,41 +81,31 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
...
+				framework.UpdateWorkload(dynamicClient, workload, "karmada-control-plane")
+			})
 
-						return memberWorkload.Spec.Paused, nil
-					}, pollTimeout, pollInterval).Should(gomega.Equal(updatedPaused))
-				}
+			ginkgo.By("check if workload's spec.paused is retained", func() {
+				framework.WaitWorkloadPresentOnClustersFitWith(framework.ClusterNames(), workload.Namespace, workload.Name,
+					func(workload *workloadv1alpha1.Workload) bool {
+						if *workload.Spec.Replicas == 2 && !workload.Spec.Paused {
```
Comment (jameszhangyukun) [inline test/e2e/resourceinterpreter_test.go:104]: directly return `*workload.Spec.Replicas == 2 && !workload.Spec.Paused`

**Comment by RainbowMango** [comment]
Found a typo on PR title.

File: test/e2e/resourceinterpreter_test.go [hunk 2]
```diff
@@ -448,30 +433,27 @@ end`,
...
+
 			ginkgo.By("check if deployment's spec.paused is retained", func() {
-				gomega.Eventually(func(g gomega.Gomega) bool {
-					var memberDeployment *appsv1.Deployment
-					framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
-						func(deployment *appsv1.Deployment) bool {
-							memberDeployment = deployment
+				framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+					func(deployment *appsv1.Deployment) bool {
+						if *deployment.Spec.Replicas == 2 && !deployment.Spec.Paused {
```
[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply test/e2e/resourceinterpreter_test.go]: Thanks~
Update.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**